presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraRecordSink.java

```diff
@@ -108,18 +117,25 @@ public void appendBoolean(boolean value)
     @Override
     public void appendLong(long value)
     {
-        if (DATE.equals(columnTypes.get(field))) {
+        Type columnType = columnTypes.get(field);
+        if (DATE.equals(columnType)) {
             append(DATE_FORMATTER.print(TimeUnit.DAYS.toMillis(value)));
         }
-        else if (INTEGER.equals(columnTypes.get(field))) {
+        else if (INTEGER.equals(columnType)) {
             append(((Number) value).intValue());
         }
-        else if (REAL.equals(columnTypes.get(field))) {
+        else if (REAL.equals(columnType)) {
             append(intBitsToFloat((int) value));
         }
-        else {
+        else if (TIMESTAMP.equals(columnType)) {
```

Comment:
Also, add tests for timestamp, making sure you're handling the value properly between Presto and Cassandra